src/pkg/cli/compose/fixup.go

```diff
@@ -153,12 +154,10 @@ func FixupServices(ctx context.Context, provider client.Provider, project *types
 			}
 		}
 
-		_, llm := svccfg.Extensions["x-defang-llm"]
-		if llm {
-			if _, ok := provider.(*client.PlaygroundProvider); ok {
-				term.Warnf("service %q: managed LLM is not supported in the Playground; consider using BYOC (https://s.defang.io/byoc)", svccfg.Name)
-				delete(svccfg.Extensions, "x-defang-llm")
-			} else if len(svccfg.Ports) == 0 {
+		if _, llm := svccfg.Extensions["x-defang-llm"]; llm {
+			image := strings.ToLower(svccfg.Image)
+			image, _, _ = strings.Cut(image, ":")
```

Comment:
could technically be hashed, ie. `alpine@sha256:a8560b36e8b8210634f77d9f7f9efd7ffa463e380b75e2e74aff4511df3ef88c` too

Reply:
TBH, let's not bother. I do like your use of `strings.Cut` so perhaps create a function that does the cut and use it elsewhere where now do `strings.SplitN(…, ":", 2)[0]`.